src/user/components/SignupPage.vue

```diff
@@ -8,7 +8,14 @@
       <input v-model="password" class="input" />
     </div>
     <div class="button">
-      <AppButton text="Sign up" @update="signUp()" />
+      <AppButton :class="'btn-primary'" content="Log in" @update="logIn()" />
+    </div>
+    <div class="button">
+      <AppButton
+        :class="'btn-secondary'"
```

Comment:
https://v3.vuejs.org/guide/class-and-style.html#object-syntax

TIL. Are there any benefits aside from dynamic CSS classes? Doesn't seem like we're using it for that purpose here

Reply:
1. my understanding is `:class` is the same as `v-bind:class` which means the class of AppButton will be determined by the string it follows
2. oof I don't think I need to wrap it with '' and "" my bad 
3. I'm not quite understanding your question is @ from @ update?
4. I think I'm using it for dynamic CSS classes...? like it will find the css class with `"btn-secondary"` and render it. Unless that's not what you're asking ahh srry

Reply:
I'm new to vue so don't have much knowledge here.
1) Makes sense, thanks!
2) All good! This is the point of having reviewers :) 

3) yeah I'm just wondering what the @ symbol does in Vue. Maybe @narendrasss knows. Can't find much in my brief search.

4) I may be missing something, but it seems like we're not actually using dynamic classes since btn-secondary seems to be provided as a string. The object syntax (`:class`) expects an object such as `:class={disabled: someFunction.shouldBeDisabled?}` so it can evaluate the class depending on the application state. 

`:class` should probably just be `class` if this is the case. This is REALLY NOT A BIG DEAL AT ALL, so dw about it! I'm new to Vue so just trying to understand the syntax 😄

Reply:
Followup: The @ notation seems to be used to link to the component methods in this case, `update` in the vue component

Reply:
@jmengo Good questions!

**Re: `:class`**

When you add the colon in front of any prop in Vue, everything that comes after it will be treated as JS syntax. For example, if you do `:class="myProp"` it will reference the `this.myProp` variable instead of the literal string `"myProp"`. The object syntax you talked about is just one out of the three valid values `:class` takes — the others being array syntax e.g. `['myClass', { 'myOtherClass': someStateIsTrue }]` and of course the string literal itself.

@michelleykim @jmengo is right here, when you only need a string you don't need the colon in front of class. I would omit it because to me it's not immediately obvious that you're using a string.

---

**Re: @ notation**

The `@some-event` notation is short for `v-on:some-event`. In the React world, think of it as a built-in `onSomeEvent` handler (i.e. you don't have to define it as a prop in the child component). In vanilla JS, it's functionally the same as `el.addEventListener('some-event', myEventHandler)`.

In React, since everything is JavaScript, you would call the `onSomeEvent` function like any other function. In Vue, you "emit" an event up the component tree using `this.$emit('some-event', eventArgs)`.

Does that make sense?